c++/src/Reader.cc

```diff
@@ -526,16 +526,25 @@ namespace orc {
     if (metadata.get() == nullptr) {
       throw std::logic_error("No stripe statistics in file");
     }
+    auto currentStripeInfo = footer->stripes(static_cast<int>(stripeIndex));
+    auto currentStripeFooter = getStripeFooter(currentStripeInfo, *contents.get());
+    const Timezone& writerTZ =
+      currentStripeFooter.has_writertimezone() ?
+        getTimezoneByName(currentStripeFooter.writertimezone()) :
+        getLocalTimezone();
+    StatContext statContext;
+    statContext.correctStats = hasCorrectStatistics();
+    statContext.writerTimezone = &writerTZ;
     return std::unique_ptr<Statistics>
       (new StatisticsImpl(metadata->stripestats
-                          (static_cast<int>(stripeIndex)),
-                          hasCorrectStatistics()));
+                          (static_cast<int>(stripeIndex)), statContext));
   }
 
   std::unique_ptr<Statistics> ReaderImpl::getStatistics() const {
+    StatContext statContext;
+    statContext.correctStats = hasCorrectStatistics();
```

Comment:
`getStatistics()` is returning the statistics in the file footer. As far as I understood, there is no timezone information available in the file footer. Looks like the timezone is stored for each stripe.
I am also curious about this design choice. I would imagine the file footer to have the timezone information. Is it possible for different stripes to have different timezone values?